src/test/scala-2/chiselTests/simulator/scalatest/ChiselSimSpec.scala

```diff
@@ -285,6 +286,95 @@ class ChiselSimSpec extends AnyFunSpec with Matchers with ChiselSim with FileChe
       }
 
     }
+
+    class RandomizationTest extends Module {
+      val in = IO(Input(UInt(32.W)))
+      val addr = IO(Input(UInt(1.W)))
+      val we = IO(Input(Bool()))
+
+      val r = IO(Output(UInt(32.W)))
+      val cm = IO(Output(UInt(32.W)))
+      val sm = IO(Output(UInt(32.W)))
+
+      private val reg = Reg(UInt(32.W))
+      private val cmem = Mem(2, UInt(32.W))
+      private val smem = SyncReadMem(2, UInt(32.W))
+
+      when(we) {
+        reg :<= in
+        cmem.write(addr, in)
+        smem.write(addr, in)
+      }
+
+      r :<= reg
+      cm :<= cmem.read(addr)
+      sm :<= smem.read(addr)
+    }
+
+    it("should have randomization on by default") {
+
+      simulate(new RandomizationTest) { dut =>
+        dut.clock.step(2)
+
+        val regValue = dut.r.peekValue().asBigInt
+        info(s"register is not zero, has value $regValue")
+        regValue should not be (0)
+
+        val cmemValue = dut.cm.peekValue().asBigInt
+        info(s"combinational read memory index zero is not zero, has value $cmemValue")
+        cmemValue should not be (0)
+
+        val smemValue = dut.sm.peekValue().asBigInt
+        info(s"sequential read memory index zero is not zero, has value $smemValue")
+        cmemValue should not be (0)
```

Comment:
I suspect the randomization is stable, but if it isn't, we have a about a 1 in 1.4 billion chance of this failing randomly every time we run this test 🙂.

Reply:
I actually wanted to increase it, but rand into a CIRCT "bug" where the randomization address is one bit too large for memories. Once I look at that, I'll increase the bit width to get the probability much lower.

Reply:
Ack, the CIRCT bug is reachable here: https://github.com/chipsalliance/chisel/actions/runs/14095137789/job/39480847729?pr=4835#step:11:1957

Reply:
I decreased the size of the `Queue` in the failing test to workaround this. I'll get a CIRCT fix in for this this week. If users run into this there are several workarounds, with the most direct being to disable that Verilator warning.

Reply:
Tracking issue here: https://github.com/llvm/circt/issues/8350